[2024-10-05T06:00:33.950+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_processing_dag.silver_cleaned_keywords manual__2024-10-05T06:00:32.300978+00:00 [queued]>
[2024-10-05T06:00:33.957+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_processing_dag.silver_cleaned_keywords manual__2024-10-05T06:00:32.300978+00:00 [queued]>
[2024-10-05T06:00:33.957+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 2
[2024-10-05T06:00:33.968+0000] {taskinstance.py:1380} INFO - Executing <Task(SparkSubmitOperator): silver_cleaned_keywords> on 2024-10-05 06:00:32.300978+00:00
[2024-10-05T06:00:33.971+0000] {standard_task_runner.py:57} INFO - Started process 5752 to run task
[2024-10-05T06:00:33.973+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spark_processing_dag', 'silver_cleaned_keywords', 'manual__2024-10-05T06:00:32.300978+00:00', '--job-id', '134', '--raw', '--subdir', 'DAGS_FOLDER/sliver_layer.py', '--cfg-path', '/tmp/tmp9chgy9x_']
[2024-10-05T06:00:33.976+0000] {standard_task_runner.py:85} INFO - Job 134: Subtask silver_cleaned_keywords
[2024-10-05T06:00:34.018+0000] {task_command.py:415} INFO - Running <TaskInstance: spark_processing_dag.silver_cleaned_keywords manual__2024-10-05T06:00:32.300978+00:00 [running]> on host 23041969bcc7
[2024-10-05T06:00:34.078+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_processing_dag' AIRFLOW_CTX_TASK_ID='silver_cleaned_keywords' AIRFLOW_CTX_EXECUTION_DATE='2024-10-05T06:00:32.300978+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-10-05T06:00:32.300978+00:00'
[2024-10-05T06:00:34.087+0000] {base.py:73} INFO - Using connection ID 'spark-conn' for task execution.
[2024-10-05T06:00:34.089+0000] {spark_submit.py:403} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension --conf spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog --jars /opt/***/jars/hadoop-aws-3.3.4.jar,/opt/***/jars/s3-2.18.41.jar,/opt/***/jars/aws-java-sdk-1.12.367.jar,/opt/***/jars/delta-core_2.13-2.1.1.jar,/opt/***/jars/delta-storage-1.2.1.jar --packages org.apache.hadoop:hadoop-aws:3.3.4,io.delta:delta-core_2.13-2.1.1 --num-executors 2 --total-executor-cores 2 --executor-cores 2 --executor-memory 2g --driver-memory 1g --name arrow-spark --deploy-mode client /opt/***/jobs/python/test_Key.py s3a://lakehouse/bronze/keywords.parquet s3a://lakehouse/silver
[2024-10-05T06:00:34.163+0000] {spark_submit.py:579} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-10-05T06:00:35.637+0000] {spark_submit.py:579} INFO - :: loading settings :: url = jar:file:/home/***/.local/lib/python3.11/site-packages/pyspark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2024-10-05T06:00:35.729+0000] {spark_submit.py:579} INFO - Exception in thread "main" java.lang.IllegalArgumentException: requirement failed: Provided Maven Coordinates must be in the form 'groupId:artifactId:version'. The coordinate provided is: io.delta:delta-core_2.13-2.1.1
[2024-10-05T06:00:35.730+0000] {spark_submit.py:579} INFO - at scala.Predef$.require(Predef.scala:281)
[2024-10-05T06:00:35.730+0000] {spark_submit.py:579} INFO - at org.apache.spark.deploy.SparkSubmitUtils$.$anonfun$extractMavenCoordinates$1(SparkSubmit.scala:1221)
[2024-10-05T06:00:35.731+0000] {spark_submit.py:579} INFO - at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
[2024-10-05T06:00:35.731+0000] {spark_submit.py:579} INFO - at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
[2024-10-05T06:00:35.732+0000] {spark_submit.py:579} INFO - at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
[2024-10-05T06:00:35.732+0000] {spark_submit.py:579} INFO - at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
[2024-10-05T06:00:35.733+0000] {spark_submit.py:579} INFO - at scala.collection.TraversableLike.map(TraversableLike.scala:286)
[2024-10-05T06:00:35.733+0000] {spark_submit.py:579} INFO - at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
[2024-10-05T06:00:35.733+0000] {spark_submit.py:579} INFO - at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:198)
[2024-10-05T06:00:35.734+0000] {spark_submit.py:579} INFO - at org.apache.spark.deploy.SparkSubmitUtils$.extractMavenCoordinates(SparkSubmit.scala:1219)
[2024-10-05T06:00:35.734+0000] {spark_submit.py:579} INFO - at org.apache.spark.deploy.SparkSubmitUtils$.resolveMavenCoordinates(SparkSubmit.scala:1541)
[2024-10-05T06:00:35.735+0000] {spark_submit.py:579} INFO - at org.apache.spark.util.DependencyUtils$.resolveMavenDependencies(DependencyUtils.scala:185)
[2024-10-05T06:00:35.735+0000] {spark_submit.py:579} INFO - at org.apache.spark.deploy.SparkSubmit.prepareSubmitEnvironment(SparkSubmit.scala:334)
[2024-10-05T06:00:35.736+0000] {spark_submit.py:579} INFO - at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:964)
[2024-10-05T06:00:35.736+0000] {spark_submit.py:579} INFO - at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:194)
[2024-10-05T06:00:35.737+0000] {spark_submit.py:579} INFO - at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:217)
[2024-10-05T06:00:35.737+0000] {spark_submit.py:579} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:91)
[2024-10-05T06:00:35.738+0000] {spark_submit.py:579} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1120)
[2024-10-05T06:00:35.738+0000] {spark_submit.py:579} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1129)
[2024-10-05T06:00:35.739+0000] {spark_submit.py:579} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2024-10-05T06:00:35.760+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension --conf spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog --jars /opt/***/jars/hadoop-aws-3.3.4.jar,/opt/***/jars/s3-2.18.41.jar,/opt/***/jars/aws-java-sdk-1.12.367.jar,/opt/***/jars/delta-core_2.13-2.1.1.jar,/opt/***/jars/delta-storage-1.2.1.jar --packages org.apache.hadoop:hadoop-aws:3.3.4,io.delta:delta-core_2.13-2.1.1 --num-executors 2 --total-executor-cores 2 --executor-cores 2 --executor-memory 2g --driver-memory 1g --name arrow-spark --deploy-mode client /opt/***/jobs/python/test_Key.py s3a://lakehouse/bronze/keywords.parquet s3a://lakehouse/silver. Error code is: 1.
[2024-10-05T06:00:35.763+0000] {taskinstance.py:1398} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_processing_dag, task_id=silver_cleaned_keywords, execution_date=20241005T060032, start_date=20241005T060033, end_date=20241005T060035
[2024-10-05T06:00:35.777+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 134 for task silver_cleaned_keywords (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension --conf spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog --jars /opt/***/jars/hadoop-aws-3.3.4.jar,/opt/***/jars/s3-2.18.41.jar,/opt/***/jars/aws-java-sdk-1.12.367.jar,/opt/***/jars/delta-core_2.13-2.1.1.jar,/opt/***/jars/delta-storage-1.2.1.jar --packages org.apache.hadoop:hadoop-aws:3.3.4,io.delta:delta-core_2.13-2.1.1 --num-executors 2 --total-executor-cores 2 --executor-cores 2 --executor-memory 2g --driver-memory 1g --name arrow-spark --deploy-mode client /opt/***/jobs/python/test_Key.py s3a://lakehouse/bronze/keywords.parquet s3a://lakehouse/silver. Error code is: 1.; 5752)
[2024-10-05T06:00:35.792+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-10-05T06:00:35.808+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
